src/nxdoc/nuxeo-server/upgrading-the-nuxeo-platform/how-to-upgrade-from-lts-2023-to-lts-2025/how-to-upgrade-to-jakarta-rs-3.md

```diff
@@ -0,0 +1,257 @@
+---
+title: How to upgrade to Jakarta RS 3 - WebEngine
+description: Instruction to upgrade your Nuxeo Project to LTS 2025 - Jakarta RS part
+review:
+   comment: ''
+   date: '2025-03-12'
+   status: ok
+labels:
+    - release-notes
+toc: true
+tree_item_index: 200
+---
+
+## Introduction
+
+{{! excerpt}}
+While the upgrade to **Jakarta RS 3** is part of the Jakarta EE 10 one, it has its own section because the implementation
+(Jersey), whose Nuxeo WebEngine is leveraging, has more changes.
+
+Nuxeo WebEngine itself has received breaking changes to benefit from the new Jakarta RS additions.
+{{! /excerpt}}
+
+You must first follow the [How to upgrade to Jakarta EE 10]({{page page='how-to-upgrade-to-jakarta-ee-10'}}) page.
+
+_JAX-RS_ occurrences have been removed from the Nuxeo Server code, it has been renamed to _REST_ where it makes sense.
+
+## Maven Dependency
+
+In addition to the Jakarta RS Specification and Implementation dependencies to change, you will need to update:
+- `nuxeo-webengine-jaxrs` to `nuxeo-webengine-rest`
+- `nuxeo-importer-jaxrs` to `nuxeo-importer-rest`
+- `nuxeo-target-platforms-jaxrs` to `nuxeo-target-platforms-rest`
+- `nuxeo-template-rendering-jaxrs` to `nuxeo-template-rendering-rest`
+
+## Java Code
+
+One of the biggest change in the Jersey Implementation is the total rework of their Dependency Injection engine.
+It is now more extendable, and better fits with the direction of Jakarta RS specification to support `jakarta.inject-api`
+annotations, such as `@Singleton` which is currently used.
+
+### Package
+
+The following Nuxeo packages have changed in order to get rid of _JAX-RS_ occurrences:
```

Comment:
`get rid` is a bit too familiar, isn't it?

Reply:
Sounds okay to me.